community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexingProvidersService.java

```diff
@@ -19,31 +19,28 @@
  */
 package org.neo4j.kernel.impl.api.index;
 
+import java.util.Optional;
+
+import org.neo4j.internal.kernel.api.schema.SchemaDescriptor;
 import org.neo4j.kernel.api.index.IndexProvider;
-import org.neo4j.kernel.api.index.IndexProvider.Descriptor;
-import org.neo4j.kernel.api.schema.index.SchemaIndexDescriptor;
+import org.neo4j.values.storable.Value;
 
-/**
- * Small class for holding a bunch of information about an index that we need to rebuild during recovery.
- */
-public class RebuildingIndexDescriptor
+public interface IndexingProvidersService
```

Comment:
I like this approach with separate interface for doing stuff that relates to specific index implementation!